src/js/components/TextInput/stories/typescript/CustomSuggestions.tsx

```diff
@@ -97,38 +120,15 @@ export const CustomSuggestions = () => {
     [],
   );
 
-  const onSuggestionsOpen = useCallback(() => setSuggestionOpen(true), []);
-  const onSuggestionsClose = useCallback(() => setSuggestionOpen(false), []);
+  const onSuggestionsOpen = useCallback(() => {
```

Comment:
Since TextInput doesn't attempt to open the drop unless there are suggestions, why do we need to `setSuggestions` here?

Reply:
The suggestions are out of date when the onChange sets value and the drop is often not open. This event gets triggered right after the onChange and since we know the drop is open, we can update the suggestions here without causing it to open. It's basically the key to the fix.